src/config.rs

```diff
@@ -307,6 +325,13 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
             .get(DNS_CAPTURE_METADATA)
             .map_or(false, |value| value.to_lowercase() == "true"),
 
+        pool_max_streams_per_conn: parse_default(POOL_MAX_STREAMS_PER_CONNECTION, 250)?,
```

Comment:
So we fail over if we happen to set this above the H2 level max stream setting. We don't have visiblity into this, so we guess.

I looked around, and defaults are:

Go: 100
Hyper: 200
Envoy: 2147483647 (lol)
Spec recommended minimum 100

This seems like if we are at 250 we are going to break even for ztunnel internal communication.

I would recommend we set this to 100.

Long term we can be adaptive to the actual limit, of course.